src/charm.py

```diff
@@ -347,7 +351,8 @@ def _environment_variables(self) -> dict:
 
     @property
     def _http_server_ip_address(self) -> Optional[IPv4Address]:
-        return IPv4Address(check_output(["unit-get", "private-address"]).decode().strip())
+        ip_address = check_output(["unit-get", "private-address"])
```

Comment:
According to the discussion from the daily, this probably shouldn't be a property anymore.